nc/api.py

```diff
@@ -77,22 +75,26 @@ def query(self, results, group_by, filter_=None):
                 purpose = PURPOSE_CHOICES.get(stop['purpose'],
                                               stop['purpose'])
                 data['purpose'] = purpose
+
             if 'person__race' in group_by:
-                race = GROUPS.get(stop['person__race'],
-                                  stop['person__race'])
-                data[race] = stop['count']
-            if 'person__ethnicity' in group_by:
-                ethnicity = GROUPS.get(stop['person__ethnicity'],
-                                       stop['person__ethnicity'])
-                data[ethnicity] = stop['count']
+                # The 'Hispanic' ethnicity option is now being aggreggated into its
+                # own race category, and its count excluded from the other counts.
+                if stop['person__ethnicity'] == 'H':
+                    race = GROUPS.get('H', 'H')
+                elif 'person__race' in group_by:
```

Comment:
It seems redundant to check for 'person__race' being in `group_by` for a second time here, no?